### DISCUSSION TITLE
How to string replace image paths when using CDN

### DISCUSSION THREAD

**Discussion by kyma** [strapi/strapi#6013] [Legacy Topics]
I've connected the upload plugin to my Azure Storage provider. It works really nicely.
I want to serve all my assets through an Azure Edge CDN.
How can I replace the image paths in my API response so that they point to the CDN location, rather than the Storage location?

**Comment by robertpiosik** [comment]
@kyma Thing is that CDN domain isn't related with strapi, what is though, is just a bucket name. So, you should set the CDN domain on the application level.

**Reply by kyma** [reply]
thought that might be the case. Just thought there might be an easy way to intercept it on the way to the DB and add a property. I'm doing it in the app and it works just fine though

**Reply by robertpiosik** [reply]
@kyma This idea was raised in #5237 and we agreed to keep it out of strapi scope.

**Comment by lauriejim** [comment]
If you want to do that in your API, you can follow this guide - https://strapi.io/documentation/3.0.0-beta.x/guides/custom-data-response.html

And instead to deleting a key, you can transform the values of your files URLs matching your need.

**Reply by kyma** [reply]
this is a nice solution, and clear how to do from the linked doc. Thanks
